docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,298 @@
+                           +----------+
+```
+
+### PersistentVolumeClaim
+
+A `PersistentVolumeClaim` does not specify which specific PersistentVolume it
+wants or even which volume plugin should be used.  Instead it specifies
+characteristics of the storage it needs and those characteristics are satisfied
+as best is possible.  Characteristics include the minimum storage space
+required and the minimum access mode required (e.g. multi-writer).
+
+As the ecosystem of volume plugins grows, there will be an ever-increasing
+number of things that could be specified about volumes.  Rather than growing
+`PersistentVolumeClaim` to cover the union of every volume's parameters, users
+will be able to request coarse "classes" of volume (e.g. Gold, Silver, Bronze).
+How those classes map to real storage technologies is a policy set by the
+cluster administrators.
+
+### VolumeManagers
+
+The act of binding a `PersistentVolumeClaim` to a `PersistentVolume` is the
+responsibility of a process in the cluster.  A cluster may have any number of
+`VolumeManager`s running, each of which handles some set of claim classes.
+When a `PersistentVolumeClaim` is created that asks for "Gold" storage, the
+`VolumeManager` which understands the "Gold" class will make the binding.
+
+`VolumeManager`s can be simple `Pod`s running in the cluster, pulling
+cofiguration from hardcoded values, flags, config objects, or whatever
+mechanism makes sense for them.  We expect config objects to be a perfect fit
+for this use case.
+
+This very loose coupling means that an admin can define classes however they
+like.  For example, "Gold" might mean GCE PersistentDisk of type SSD, "Silver"
+might mean GCE PersistentDisk of type Disk, and "Bronze" might mean NFS.
+Configuring the same `VolumeManager` to run for both "Gold" and "Silver" with
+just a change in params is trivial, as would be adding a totally bespoke
+`VolumeManager`.
+
+Note that classes are just identifiers - there are no special names or
+meanings.  One of the `VolumeManager`s must be the "default" for claims that do
+not specify a class.
+
+### Config
+
+We expect that `VolumeManager`s will need some amount of configuration, and it
+will be convenient if they have similar parameters, but this should be by
+convention, not by diktat.  Example parameters:
+   * Minimum free instances - how many ready-to-use volumes should be maintained
+   * Maximum free instances - limit of number of ready-to-use volumes
+   * Size distribution - desired histogram of sizes of ready-to-use volumes
+
+In addition to these, there will be volume-type specific parameters, such as
+the GCE SSD vs Disk param described above.
+
+### Scrubbing
+
+Volumes need to be scrubbed before they can be reused, and some storage devices
+may even need to be scrubbed before they are decommissioned.  This is the
+responsibility of the `VolumeManager` as it drives the lifecycle.
+
+## Networking
+
+TODO: flip Ingress and LB ?
```

Comment:
@bprashanth I am unsatisfied with the naming after writing this up.

In volumes: Pod -> PVclaim -> PV

In network: Ingress -> ???Claim -> LoadBalancer (as we have sketched it).  By convention that means "???Claim" is  a LoadBalancerClaim - are we OK with that?

Or should it be flipped:  user creates a LoadBalancer -> IngressClaim -> Ingress ?

Or something else?  What layer actually knows the difference between a generic ingress mode and a load-balancer?  Are there other kinds of Ingress that are not load-balancers (e.g. VM IPs in GCE)